src/openms/source/KERNEL/MSSpectrum.cpp

```diff
@@ -317,6 +322,49 @@ namespace OpenMS
     return (max_intensity_it - this->begin());
   }
 
+
+  void MSSpectrum::sortByPositionPresorted(const std::vector<std::pair<Size, bool>>& chunks)
+  {
+    if (isSorted()) return;
```

Comment:
this is never true, is it?

Reply:
unless chunks.size()==1 and the chunk is already sorted

Reply:
Let's say, almost never. There might be cases depending on what Ion-Types are chosen.

Reply:
I can take it out anyway, because the overhead of the new sort might not be too bad on a sorted vector, compared to the isSorted() on every unsorted vector.